### PR TITLE
Ranks

### PR DESCRIPTION
**Pull request by peter-budo**
- Adding rank image resources
- Added map of rank image resources and updated soldier overview rank image
- Extracting Java hardcoded string values to string resources. This way any translator can translate ranks too. We should try to avoid hard coding stuff like it was originally.
- Must first setMax on progress bar and only after it progress or it will not work properly.
- Display service stars in order

### CODE REVIEW

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankImageMap.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
+package com.ninetwozero.bf4intel.resources.maps;
+
+import com.ninetwozero.bf4intel.R;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class RankImageMap {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankImageMap.java:8]: :+1:

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankStringMap.java [hunk 2]
```diff
@@ -1,116 +1,118 @@
 package com.ninetwozero.bf4intel.resources.maps;
 
+import com.ninetwozero.bf4intel.R;
+
 import java.util.HashMap;
 import java.util.Map;
 
 public class RankStringMap {
-    private static final Map<String, String> map = new HashMap<String, String>() {
+    private static final Map<String, Integer> map = new HashMap<String, Integer>() {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankStringMap.java:9]: Just a spontaneous question - do we really want to have them translatable? I would like to think that they are somewhat "universal", but that may not be the case. :-)

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankStringMap.java [hunk 3]
```diff
@@ -1,116 +1,118 @@
...
+            put("WARSAW_ID_P_RANK98_NAME", R.string.rank98);
+            put("WARSAW_ID_P_RANK99_NAME", R.string.rank99);
+            put("WARSAW_ID_P_RANK100_NAME", R.string.rank100);
         }
     };
 
-    public static String get(final String key) {
-        return map.containsKey(key)? map.get(key) : key;
+    public static int get(final String key) {
+        return map.containsKey(key)? map.get(key) : -1;
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankStringMap.java]: -1 is an invalid value - default it to R.string.na or something like that. Shortened error below:

```
 android.content.res.Resources$NotFoundException: String resource ID #0xffffffff
```

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankStringMap.java (hunk 2 above)
Comment (peter-budo) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankStringMap.java:9]: You sure? :smiling_imp: Quick check. Battlelog, scroll end of page, change language Czech and my rank is `Starší nadrotmistr II`

Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/RankStringMap.java:9]: Then :+1:

**Comment by karllindmark** [comment]
By the way @peter-budo, is the red circle on the rank images from Dice themselves or did we add it to them? They look a bit fuzzy in-app. Check out the attached image for a better explanation: http://puu.sh/5zrP4.png

**Comment by peter-budo** [comment]
I'm looking into it, no clear reason why they look differently between web and local download

**Comment by karllindmark** [comment]
I think I have an idea - how did you save the images? If you save them without transparency, then the "red mist" becomes a solid blob. Does that seem like a plausible idea?

**Pull request merged**